src/include/duckdb/function/cast/variant/to_variant_fwd.hpp

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/common/types/variant.hpp"
+#include "duckdb/common/types/selection_vector.hpp"
+#include "duckdb/common/owning_string_map.hpp"
+#include "duckdb/common/optional_ptr.hpp"
+#include "duckdb/common/serializer/varint.hpp"
+#include "duckdb/common/types/decimal.hpp"
+#include "duckdb/common/exception/conversion_exception.hpp"
+
+namespace duckdb {
+namespace variant {
+
+struct OffsetData {
+public:
+	static uint32_t *GetKeys(DataChunk &offsets) {
+		return FlatVector::GetData<uint32_t>(offsets.data[0]);
+	}
+	static uint32_t *GetChildren(DataChunk &offsets) {
+		return FlatVector::GetData<uint32_t>(offsets.data[1]);
+	}
+	static uint32_t *GetValues(DataChunk &offsets) {
+		return FlatVector::GetData<uint32_t>(offsets.data[2]);
+	}
+	static uint32_t *GetBlob(DataChunk &offsets) {
+		return FlatVector::GetData<uint32_t>(offsets.data[3]);
+	}
+};
+
+struct VariantVectorData {
```

Comment:
Consider making this public, in `variant.hpp`, instead of making this an internal helper just for the cast functions